lib/membrane_vpx/encoder/vp8_encoder.ex

```diff
@@ -7,13 +7,15 @@ defmodule Membrane.VP8.Encoder do
   alias Membrane.{VP8, VPx}
 
   def_options encoding_deadline: [
-                spec: non_neg_integer(),
-                default: 1,
+                spec: Membrane.Time.t() | :auto,
+                default: :auto,
                 description: """
-                Determines how long should it take the encoder to encode a frame (in microseconds).
+                Determines how long should it take the encoder to encode a frame.
                 The longer the encoding takes the better the quality will be. If set to 0 the
                 encoder will take as long as it needs to produce the best frame possible. Note that
                 this is a soft limit, there is no guarantee that the encoding process will never exceed it.
+                If set to `:auto` the deadline will be calculated based on the framerate provided by
+                incoming stream format. If it's `nil` a fixed deadline of 10ms will be set.
```

Comment:
```suggestion
                incoming stream format. If the framerate is `nil`, a fixed deadline of 10ms will be set.
```

Reply:
(I had some trouble figuring out what does it mean as I thought that it's possible to set `nil` as `encoding_deadline` option :D )